src/cs50/_statement.py

```diff
@@ -0,0 +1,279 @@
+                sqlparse.tokens.String, string_processor(value.strftime("%Y-%m-%d %H:%M:%S")))
+
+        if isinstance(value, datetime.time):
+            return sqlparse.sql.Token(
+                sqlparse.tokens.String, string_processor(value.strftime("%H:%M:%S")))
+
+        if isinstance(value, float):
+            return sqlparse.sql.Token(
+                sqlparse.tokens.Number,
+                sqlalchemy.types.Float().literal_processor(self._dialect)(value))
+
+        if isinstance(value, int):
+            return sqlparse.sql.Token(
+                sqlparse.tokens.Number,
+                sqlalchemy.types.Integer().literal_processor(self._dialect)(value))
+
+        if isinstance(value, str):
+            return sqlparse.sql.Token(sqlparse.tokens.String, string_processor(value))
+
+        if value is None:
+            return sqlparse.sql.Token(
+                sqlparse.tokens.Keyword,
+                sqlalchemy.types.NullType().literal_processor(self._dialect)(value))
+
+        raise RuntimeError(f"unsupported value: {value}")
+
+
+    def _escape_iterable(self, iterable):
+        return sqlparse.sql.TokenList(
+            sqlparse.parse(", ".join([str(self._escape(v)) for v in iterable])))
+
+
+    def get_command(self):
+        """Returns statement command (e.g., SELECT) or None"""
+        return self._command
+
+
+    def __str__(self):
+        return "".join([str(token) for token in self._tokens])
+
+
+def _is_placeholder(token):
+    return token.ttype == sqlparse.tokens.Name.Placeholder
+
+
+def _parse_placeholder(token):
+    if token.value == "?":
+        return _Paramstyle.QMARK, None
+
+    # E.g., :1
+    matches = re.search(r"^:([1-9]\d*)$", token.value)
+    if matches:
+        return _Paramstyle.NUMERIC, int(matches.group(1)) - 1
+
+    # E.g., :foo
+    matches = re.search(r"^:([a-zA-Z]\w*)$", token.value)
+    if matches:
+        return _Paramstyle.NAMED, matches.group(1)
+
+    if token.value == "%s":
+        return _Paramstyle.FORMAT, None
+
+    # E.g., %(foo)s
+    matches = re.search(r"%\((\w+)\)s$", token.value)
+    if matches:
+        return _Paramstyle.PYFORMAT, matches.group(1)
+
+    raise RuntimeError(f"{token.value}: invalid placeholder")
+
+
+def _escape_verbatim_colons(tokens):
+    for token in tokens:
+        if _is_string_literal(token):
+            token.value = re.sub(r"(^'|\s+):", r"\1\:", token.value)
+        elif _is_identifier(token):
+            token.value = re.sub(r"(^\"|\s+):", r"\1\:", token.value)
+
+    return tokens
+
+
+def _is_command_token(token):
+    return token.ttype in {
+        sqlparse.tokens.Keyword, sqlparse.tokens.Keyword.DDL, sqlparse.tokens.Keyword.DML}
+
+
+def _is_string_literal(token):
+    return token.ttype in [sqlparse.tokens.Literal.String, sqlparse.tokens.Literal.String.Single]
+
+
+def _is_identifier(token):
+    return token.ttype == sqlparse.tokens.Literal.String.Symbol
+
+
+class _Paramstyle(enum.Enum):
```

Comment:
What's this do?

Reply:
Just an enum of the different paramstyle types that are supported so we don't have to hardcode `"qmark"`, `"numeric"`, etc and if we add or change types, their names are checked for (e.g., for misspellings).